core/shared/src/main/scala/org/http4s/MediaType.scala

```diff
@@ -410,4 +415,10 @@ object MediaType extends MimeDB {
         }
       case _ => false
     }
+
+  // lazy, incomplete predictor, used only on JS
+  private[this] def isCompressible(mainType: String, subType: String): Boolean = {
+    val _ = subType
+    mainType == "text"
+  }
```

Comment:
AFAICT all `text` types are compressible. We can add more cases as needed, but I don't expect this to get hit much outside of our tests since this is used only for the server middleware.